source/opt/optimizer.cpp

```diff
@@ -169,7 +169,8 @@ Optimizer& Optimizer::RegisterLegalizationPasses(bool preserve_interface) {
           .RegisterPass(CreateAggressiveDCEPass(preserve_interface))
           .RegisterPass(CreateRemoveUnusedInterfaceVariablesPass())
           .RegisterPass(CreateInterpolateFixupPass())
-          .RegisterPass(CreateInvocationInterlockPlacementPass());
+          .RegisterPass(CreateInvocationInterlockPlacementPass())
+          .RegisterPass(CreateOpExtInstWithForwardReferenceFixupPass());
```

Comment:
Is there a reason you put it at the end instead of at the start? It is important that other optimizations run first?

Reply:
I wanted to make sure any optimization modifying the code order was running first (thus we would have the final SPIR-V layout, and correctly fix OpExtInst/OpExtInstWithForward reference)